crates/next-core/js/src/internal/edge.ts

```diff
@@ -0,0 +1,143 @@
+import { join } from "path";
```

Comment:
Link which file in `next.js` you adapted for this code, so we can more easily keep track of updates/origin code.